build.gradle

```
@@ -58,6 +58,17 @@ task copyJars(type: Copy) {
 }
 compileJava.dependsOn(copyJars)
 
+task mainJar(type: Jar, dependsOn: classes) {
+    baseName = project.name + '-main'
```

```suggestion
    archiveBaseName = project.name + '-main'
```

`$ ./gradlew --warning-mode all compileJava`

```
> Configure project :
The AbstractArchiveTask.baseName property has been deprecated. This is scheduled to be removed in Gradle 8.0. Please use the archiveBaseName property instead. See https://docs.gradle.org/7.4.2/dsl/org.gradle.api.tasks.bundling.AbstractArchiveTask.html#org.gradle.api.tasks.bundling.AbstractArchiveTask:baseName for more details.
        at build_eynxkc02z8az0l7b1w8zg2x5g$_run_closure4.doCall(/RapidWright/build.gradle:62)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)
```